server/controllers/medical/reports/visits.report.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Visits Report
+ */
+const _ = require('lodash');
+const moment = require('moment');
+const ReportManager = require('../../../lib/ReportManager');
+const db = require('../../../lib/db');
+
+// path to the template to render
+const TEMPLATE = './server/controllers/medical/reports/visits.report.handlebars';
+
+exports.document = document;
+
+function document(req, res, next) {
+  let report;
+  let details;
+
+  const params = req.query;
+
+  const optionReport = _.extend(params, {
+    filename : 'PATIENT_RECORDS.REPORT.VISITS',
+  });
+
+  try {
+    details = req.query.params ? JSON.parse(req.query.params) : {};
+    report = new ReportManager(TEMPLATE, req.session, optionReport);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  getData(details)
+    .then(visits => report.render({ visits, details }))
+    .then(result => res.set(result.headers).send(result.report))
+    .catch(next)
+    .done();
+}
+
+function getData(options) {
+  const baseQuery = `
```

Comment:
Wee!  This is a lot of queries.  I think we can combine most of them as `SELECT SUM(IF(condition, 1, 0))`.  Let's try:

```sql
SELECT
  s.name,
  SUM(1) AS total,
  SUM(IF(is_new_case, 1, 0)) AS total_new_case,
  SUM(IF(is_new_case, 0, 1)) AS total_old_case,
  SUM(IF(is_pregnant, 1, 0)) AS total_pregnant,
  SUM(IF(is_refered, 1, 0)) AS total_referred,
  SUM(IF(is_hospitalized, 1, 0)) AS total_hospitalized,
  SUM(IF(inside_health_zone, 1, 0)) AS total_inside_health_zone,
  SUM(IF(inside_health_zone = 0, 1, 0)) AS total_outside_health_zone
FROM service s 
  LEFT JOIN patient_visit_service pvs ON s.id = pvs.service_id
  JOIN patient_visit ON pvs.patient_visit_uuid = pv.uuid 
GROUP BY s.id
WHERE pvs.created_at BETWEEN DATE(?) AND DATE(?));
```

I think that should get you pretty close! One SQL query is better than many in this case :).